bin/check-locations

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+"""
+Validates the locations in the metadata TSV file produced by the transform
+scripts. If the validations succeed, saves a location hierarchy from the
+given metadata to a local file.
+"""
+import sys
+import argparse
+import pandas as pd
+from pathlib import Path
+from typing import List
+
+sys.path.insert(0, str(Path(__file__).parent.parent / "lib"))
+from utils.hierarchy_dataframe import hierarchy_dataframe
+
+LOCATION_HIERARCHY_COLUMNS = ['region', 'country', 'division']
+
+base = Path(__file__).resolve().parent.parent
+
+
+def validate_metadata(metadata: pd.DataFrame, columns: List[str]):
+    def no_missing_hierarchy_data():
+        """
+        Raises an ``AssertionError`` if the metadata have missing values in the
+        given location hierarchy *columns*.
+        """
+        assert metadata[columns].notnull().all().all(), \
+            "There are missing values in the location hierarchy columns " \
+            f"«{columns}». At this point, we expect all " \
+            "missing values to be interpolated."
+
+    no_missing_hierarchy_data()
```

Comment:
So the inner function here is a bit of overkill because it's the only validation being run. But, I separated the `no_missing_hierarchy_data()` logic into its own function because future PRs will add more functionality to `validate_metadata()`. 

I defined the function inside of `validate_metadata()` because the larger scope of this script doesn't need to know about `no_missing_hierarchy_data()`. Really, defining it as an inner function is really just a personal preference. When looking at the `validate_metadata()` function, I find it easier to read all the validations that will be run within the definition of `validate_metadata()` than to go looking for the function definitions elsewhere in the script.